plugins/scale/wayfire/plugins/scale-signal.hpp

```diff
@@ -73,4 +74,32 @@ void scale_filter_views(scale_filter_signal *signal, pred&& p)
  * argument: unused
  */
 
+/**
+ * name: scale-transformer-added
+ * on: output
+ * when: This signal is emitted when scale adds a transformer to a view, so
+ *   plugins extending its functionality can add their overlays to it.
+ * argument: pointer to the newly added transformer
+ */
+struct scale_transformer_added_signal : public wf::signal_data_t
+{
+    wf::scale_transformer *transformer;
+};
+
+/**
+ * name: scale-padding
```

Comment:
Looking at it again, don't we already have the padding in the transformer? Why do we need to calculate it once more here?

Reply:
The main reason for this is that calculating the layout might happen before the transformers are initialized, so scale needs to know about padding before that.
But this means that we are supplying the padding essentially twice.
Now I'm leaning toward keeping the signal, and making it scale's responsibility to add padding to the transformer. Overlays are then not allowed to extend padding (at least without triggering an update to scale that will emit this signal again). I'll add a version with this implimented.

Reply:
> The main reason for this is that calculating the layout might happen before the transformers are initialized, so scale needs to know about padding before that.

It all depends on how we implement it. We could compute the actual view size after adding the transformer (which is the moment overlays are added too).